README.md

```diff
@@ -1,22 +1,47 @@
 # ansible-mps-plugin
+The ansible-mps-plugin is one of many plugins of the [DeMAF](https://github.com/UST-DeMAF) project.
+It is designed to transform [Ansible deployment models](https://docs.ansible.com) into an [EDMM](https://github.com/UST-EDMM) representation.
 
-## Build
-```shell
-./mvnw package
-```
+The plugin only works (without adaptions) in the context of the entire DeMAF application using the [deployment-config](https://github.com/UST-DeMAF/deployment-config).
+The documentation for setting up the entire DeMAF application locally is [here](https://github.com/UST-DeMAF/EnPro-Documentation).
+
+## Build and Run Application
+
+You can run the application without the [deployment-config](https://github.com/UST-DeMAF/deployment-config) but it will not run as it needs to register itself at the [analysis-manager](https://github.com/UST-DeMAF/analysis-manager).
+
+If you want to boot it locally nevertheless use the following commands.
 
-## Run
 ```shell
 ./mvnw spring-boot:run
 ```
-or (requires build):
+or:
 ```shell
+./mvnw package
 java -jar target/ansible-plugin-0.0.1-SNAPSHOT.jar
 ```
 
-## Update Submodule
-The JetBrains MPS application that executes the transformation is added as a git submodule.
-To update it to a new version, execute:
+## Init and Update Submodule
+This plugin uses [JetBrains MPS](https://www.jetbrains.com/mps/) to facilitate the model-to-model transformation from Ansible to EDMM.
+The [matching MPS project](https://github.com/UST-DeMAF/mps-transformation-terraform) is located in another git repository and must be added as a submodule (you can also clone via https):
+
+```shell
+git submodule add git@github.com:UST-DeMAF/mps-transformation-ansible.git mps-transformation-terraform
+```
+
+To update the MPS application to a new version, execute:
 ```shell
 git submodule update --remote
-```
\ No newline at end of file
+```
+
+## Ansible-Specific Configurations
+This plugin has some minor specialities compared to other DeMAF transformation plugins:
+1. No directories: Ansible projects are created in a folder structure, where it is possible to have multiple overlapping main files.
+Therefore, it is not possible to scan directories, but always only one main file.
+Of course, matching subdirectories, e.g. for roles are possible to read.
```

Comment:
Maybe add this to the TODO.md?

Reply:
this is not a todo, but actually by design, I already discussed it with Marcel :)